build_metal.sh

```diff
@@ -74,105 +39,169 @@ enable_tsan="OFF"
+
+OPTIONS=h,e,c,t,a,m,s,u,b:,p
+LONGOPTS=help,export_compile_commands,enable_ccache,enable_time_trace,enable_asan,enable_msan,enable_tsan,enable_ubsan,build-type:,release,development,debug,enable_profiler,build-tests,install-prefix:,cmake-options:,clean
+
+# Parse the options
+PARSED=$(getopt -n 'build_metal' --options=$OPTIONS --longoptions=$LONGOPTS --name "$0" -- "$@")
+if [[ $? -ne 0 ]]; then
+    # If getopt has errors
+    echo "INFO: Failed to parse arguments!"
+    show_help
+    exit 1
+fi
+
+eval set -- "$PARSED"
+
+while true; do
+    case "$1" in
+        -h|--help)
             show_help
             exit 0
             ;;
-        e )
-            export_compile_commands="ON"
-            ;;
-        c )
-            enable_ccache="ON"
-            ;;
-        t )
-            enable_time_trace="ON"
-            ;;
-        a )
-            enable_asan="ON"
-            ;;
-        m )
-            enable_msan="ON"
-            ;;
-        s )
-            enable_tsan="ON"
-            ;;
-        u )
-            enable_ubsan="ON"
-            ;;
-        b )
-            build_type="$OPTARG"
-            ;;
-        p )
-            enable_profiler="ON"
-            ;;
-        \? )
-            show_help
-            exit 1
-            ;;
+        -e|--export-compile-commands)
+            export_compile_commands="ON";;
+        -c|--enable-ccache)
+            enable_ccache="ON";;
+        -t|--enable-time-trace)
+            enable_time_trace="ON";;
+        -a|--enable-asan)
+            enable_asan="ON";;
+        -m|--enable-msan)
+            enable_msan="ON";;
+        -s|--enable-tsan)
+            enable_tsan="ON";;
+        -u|--enable-ubsan)
+            enable_ubsan="ON";;
+        -b|--build-type)
+            echo "Manually specifying build type"
+            build_type="$2";shift;;
+        -rel|--release)
+            build_type="Release";;
+        -dev|--development)
+            build_type="RelWithDebInfo";;
+        -dbg|--debug)
+            build_type="Debug";;
+        -p|--enable-profiler)
+            enable_profiler="ON";;
+        --build-tests)
+            build_tests="ON";;
+        --install-prefix)
+            install_prefix="$2";shift;;
+        --cmake-options)
+	    cmake_args+=($2);shift;;
+        --clean)
+	    clean; exit 0;;
+        --)
+            shift;break;;
     esac
+    shift
 done
 
+# Check if there are unrecognized positional arguments left
+if [[ $# -gt 0 ]]; then
+    echo "ERROR: Unrecognized positional argument(s): $@"
+    show_help
+    exit 1
+fi
+
+# Validate the build_type
+VALID_BUILD_TYPES=("Release" "Debug" "RelWithDebInfo")
+if [[ ! " ${VALID_BUILD_TYPES[@]} " =~ " ${build_type} " ]]; then
+    echo "ERROR: Invalid build type '$build_type'. Allowed values are Release, Debug, RelWithDebInfo."
+    show_help
+    exit 1
+fi
+
+build_dir="build_$build_type"
+
+
 # Set the python environment directory if not already set
 if [ -z "$PYTHON_ENV_DIR" ]; then
```

Comment:
Is this needed in the CMake system? Don't think so right?

Reply:
Not sure actually.